test/MessageTraitTest.php

```diff
@@ -66,6 +65,27 @@ public function testWithProtocolVersionRaisesExceptionForInvalidVersion($version
         $request->withProtocolVersion($version);
     }
 
+    public function validProtocolVersionProvider()
+    {
+        return [
+            '1.0' => [ '1.0' ],
+            '1.1' => [ '1.1' ],
+            '2'   => [ '2' ],
+            '2.0' => [ '2.0' ],
+        ];
+    }
+
+    /**
+     * @dataProvider validProtocolVersionProvider
+     */
+    public function testWithProtocolVersionDoesntRaiseExceptionForValidVersion($version)
```

Comment:
Parameter/return type is to be provided

Reply:
These don't exist on any method within this class which is why I didn't add it. I'll go through and add to all the methods in this class for consistency if it's required for my changes